test/parser.cpp

```diff
@@ -405,6 +405,61 @@ R"xx({
                 .as_int64() == 12345);
     }
 
+    void
+    testIssue38()
+    {
+        {
+            auto expected = string_view("Hello\x00World", 11);
+            {
+                auto s = string_view(R"json("Hello\u0000World")json");
+                grind(s);
+                auto v = json::parse(s);
+                if (BOOST_TEST(v.is_string()))
+                {
+                    auto s2 = v.as_string();
+                    BOOST_TEST(s2 == expected);
+                }
+            }
+            {
+                auto s = string_view(R"json(["Hello\u0000World"])json");
+                grind(s);
+                auto v = json::parse(s);
+                BOOST_ASSERT(v.is_array());
+                auto a = v.as_array();
+                BOOST_ASSERT(a.size() == 1);
+                auto v2 = a[0];
+                BOOST_ASSERT(v2.is_string());
+                auto s2 = v2.as_string();
+                BOOST_TEST(s2 == expected);
```

Comment:
This is a pretty verbose way of simply saying
```
BOOST_TEST( json::parse(s).as_array()[0].as_string() == expected );
```

Reply:
Or even
```
BOOST_TEST(
    json::parse(string_view( "[\"Hello\u0000World\"]" ))
        .as_array().at(0).as_string() == string_view("Hello\x00World", 11) );
```

Reply:
made less verbose.
Renamed the test case to indicate function rather than motivation